<launch>
    <arg name="robot_name"  default="robot_0"/>
    <arg name="laser_topic"  default="front_scan"/>
    <arg name="odom_topic"  default="odom"/>
    <arg name="depth_registration" default="true"/>
    <arg name="color_depth_synchronization" default="true"/>

    <param name="robotname" value="$(arg robot_name)" />
    
    <group ns="$(arg robot_name)">
      <include file ="$(find openni2_launch)/launch/openni2.launch">
	<arg name="depth_registration" value="$(arg depth_registration)" />
        <arg name="color_depth_synchronization" value="$(arg color_depth_synchronization)" />
	
      </include>
     
      <node pkg="dot_detector" type="dot_detector" name="dot_detector" output="screen">
          <param name="log_dir_path" value="$(find dot_detector)/images_log" />
          <param name="tf_prefix" value="$(arg robot_name)" />
          <param name="laser_topic" value="/$(arg robot_name)/$(arg laser_topic)" />
          <param name="odom_topic" value="/$(arg robot_name)/$(arg odom_topic)" />
          <param name="robot_name" value="$(arg robot_name)" />

	<param name="top_camera_name" value="/$(arg robot_name)/camera" />
	
      </node>
    </group>
    
    <node name="objects_processing" pkg="objects_processing" type="objects_processing" args="0" output="screen" />

</launch>

